packages/radio-group/src/Radio.tsx

```diff
@@ -86,70 +226,77 @@ export interface RadioProps {
  * @param props.id Id for Radio and respective label.
  * @param props.default If RadioGroup is uncontrolled, the default property makes this Radio checked on the initial render.
  * @param props.className className supplied to Radio container.
+ * @param props.size Size of Radio buttons.
  */
-export default class Radio extends PureComponent<
-  RadioProps & React.InputHTMLAttributes<HTMLInputElement>
-> {
-  static displayName = 'Radio';
-
-  static propTypes = {
-    checked: PropTypes.bool,
-    disabled: PropTypes.bool,
-    className: PropTypes.string,
-    children: PropTypes.node,
-    onChange: PropTypes.func,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
-    id: PropTypes.string,
-    name: PropTypes.string,
-    variant: PropTypes.oneOf(['default', 'light']),
-    default: PropTypes.bool,
-  };
-
-  static defaultProps = {
-    checked: false,
-    disabled: false,
-    className: '',
-    onChange: () => {},
-    variant: Variant.Default,
-  };
-
-  render() {
-    const {
-      children,
-      className,
-      onChange,
-      value,
-      checked,
-      disabled,
-      id,
-      name,
-      variant,
-      ...rest
-    } = this.props;
-
-    return (
-      <label htmlFor={id} className={cx(labelStyle, className)}>
+function Radio({
+  children,
+  className,
+  onChange,
+  value,
+  disabled,
+  id,
+  name,
+  checked = false,
+  size = Size.Default,
+  variant = Variant.Default,
+  ...rest
+}: RadioProps) {
+  return (
+    <div className={containerMargin}>
+      <label
+        htmlFor={id}
+        className={cx(
+          labelStyle,
+          labelVariantStyle[variant].base,
+          { [labelVariantStyle[variant].disabled]: disabled },
+          className,
+        )}
+      >
         <input
           {...rest}
           id={id}
           name={name}
           type="radio"
-          className={inputStyle}
           onChange={onChange}
           value={value}
-          checked={checked}
           aria-checked={checked}
           disabled={disabled}
           aria-disabled={disabled}
+          className={cx(inputStyle, inputVariantStyle[variant], {
+            [disabledChecked[variant]]: disabled && checked,
+          })}
         />
-        <span
-          className={cx(textStyle, {
-            [disabledButtonVariant[variant]]: disabled,
+        <div
+          {...styledDiv.prop}
+          className={cx(divStyle, divVariantStyle[variant], getDivHeight(size))}
```

Comment:
How come we're using an object for `divVariantStyle`, but a function for `getDivHeight`? Is there a reason that we couldn't rework this so that you could do `divSizeStyle[size]` to be consistent?

Reply:
👍